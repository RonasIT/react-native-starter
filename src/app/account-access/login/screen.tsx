import { yupResolver } from '@hookform/resolvers/yup';
import Constants from 'expo-constants';
import { isEmpty } from 'lodash';
import React from 'react';
import { useForm } from 'react-hook-form';
import { ScrollView, View } from 'react-native';
import { Image, Keyboard } from 'react-native-ui-lib';
import { assets } from '@assets/index';
import { AppVersion } from '@shared/app-version';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText } from '@shared/text';
import { commonStyle, createStyles } from '@styles';
import { loginScreenFacade } from './facade';
import { LoginForm } from './shared/forms';

export function LoginScreen(): JSX.Element {
  const translate = useTranslation('ACCOUNT_ACCESS.LOGIN');
  const { isSubmitting } = loginScreenFacade;
  const appName = Constants.expoConfig.name;

  function formSubmitted(values: LoginForm): void {
    loginScreenFacade.authorize(values);
  }

  const form = useForm<LoginForm>({
    defaultValues: new LoginForm(),
    resolver: yupResolver(LoginForm.validationSchema)
  });
  const { handleSubmit, formState, control } = form;

  return (
    <AppScreen style={commonStyle.container}>
      <ScrollView contentContainerStyle={style.content}>
        <Image source={assets.brand.logo} style={style.logo} />
        <AppText style={style.title} variant='largest'>
          {translate('TEXT_TITLE', { value: appName })}
        </AppText>
        <InputFormGroup<LoginForm>
          label={translate('TEXT_LOGIN')}
          name='email'
          testID='email-input'
          autoCapitalize='none'
          keyboardType='email-address'
          control={control}
        />
        <InputFormGroup<LoginForm>
          isPassword={true}
          testID='password-input'
          label={translate('TEXT_PASSWORD')}
          name='password'
          control={control}
        />
        <View style={style.footer}>
          <AppButton
            isLoading={isSubmitting}
            disabled={!isEmpty(formState.errors) && formState.isSubmitted}
            testID='submit-button'
            label={translate('BUTTON_SUBMIT')}
            onPress={handleSubmit(formSubmitted)}
          />
        </View>
        <AppVersion />
        <Keyboard.KeyboardAwareInsetsView />
      </ScrollView>
    </AppScreen>
  );
}

const style = createStyles({
  content: {
    paddingTop: '2rem'
  },
  logo: {
    width: '3.5rem',
    height: '3.5rem'
  },
  title: {
    marginVertical: '2rem'
  },
  footer: {
    marginTop: '2rem'
  }
});
