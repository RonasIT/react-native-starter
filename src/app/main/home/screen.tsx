import React, { ReactElement } from 'react';
import { AppScreen } from '../../../libs/shared/ui/ui-kit/screen';
import { UsersList } from '../../../libs/users/features/list';

export function HomeScreen(): ReactElement {
  return (
    <AppScreen testID='home-screen'>
      <UsersList />
    </AppScreen>
  );
}
