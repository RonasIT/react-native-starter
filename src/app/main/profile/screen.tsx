import { AppActivityIndicator } from '@shared/activity-indicator';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { AppScreen } from '@shared/screen';
import { AppText, TextTheme } from '@shared/text';
import { commonStyle, createStyles } from '@styles';
import React, { ReactElement } from 'react';
import { profileScreenFacade } from './facade';

export function ProfileScreen(): ReactElement {
  const { profile, logout } = profileScreenFacade;
  const translate = useTranslation('MAIN.PROFILE');

  return (
    <AppScreen style={[commonStyle.container, style.container]}>
      {profile ? (
        <AppText theme={TextTheme.LARGEST}>{translate('TEXT_GREETING', { name: profile?.name })}</AppText>
      ) : (
        <AppActivityIndicator />
      )}
      <AppButton title={translate('BUTTON_LOGOUT')} onPress={logout} />
    </AppScreen>
  );
}

const style = createStyles({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-evenly'
  }
});
