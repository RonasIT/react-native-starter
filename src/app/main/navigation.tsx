import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { useTranslation } from '@shared/i18n';
import { Icon } from '@shared/icon';
import { AppTabBar } from '@shared/tab-bar';
import { variables } from '@styles';
import React, { ReactElement } from 'react';
import { HomeScreen } from './home/screen';
import { ProfileScreen } from './profile/screen';

export type MainNavigationParams = {
  Home: undefined;
  Profile: undefined;
};

const Tab = createBottomTabNavigator<MainNavigationParams>();

export function MainNavigation(): ReactElement {
  const initialRouteName = 'Home';
  const translate = useTranslation('MAIN.NAVIGATION');

  return (
    <Tab.Navigator
      tabBar={(props) => <AppTabBar {...props} />}
      initialRouteName={initialRouteName}
      backBehavior='none'>
      <Tab.Screen
        name='Home'
        options={{
          tabBarIcon: ({ focused }) => (
            <Icon name='home' stroke={focused ? variables.color.primary : variables.color.white} />
          ),
          headerTitleAlign: 'left',
          title: translate('TEXT_HOME')
        }}
        component={HomeScreen}
      />
      <Tab.Screen
        name='Profile'
        options={{
          tabBarIcon: ({ focused }) => (
            <Icon name='user' stroke={focused ? variables.color.primary : variables.color.white} />
          ),
          headerTitleAlign: 'left',
          title: translate('TEXT_PROFILE')
        }}
        component={ProfileScreen}
      />
    </Tab.Navigator>
  );
}
