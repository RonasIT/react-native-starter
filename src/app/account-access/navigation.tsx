import { createStackNavigator } from '@react-navigation/stack';
import React, { ReactElement } from 'react';
import { appFacade } from '@app/facade';
import { useTranslation } from '@shared/i18n';
import { LoginScreen } from './login/screen';
import { PinScreen } from './pin';

export type AccountAccessNavigationParams = {
  Login: undefined;
  Pin: { isPinSet?: boolean };
};

const Stack = createStackNavigator<AccountAccessNavigationParams>();

export function AccountAccessNavigation(): ReactElement {
  const translate = useTranslation('ACCOUNT_ACCESS.NAVIGATION');
  const { isAuthenticated } = appFacade;

  return (
    <Stack.Navigator initialRouteName={isAuthenticated ? 'Pin' : 'Login'}>
      <Stack.Screen
        initialParams={{ isPinSet: isAuthenticated }}
        options={{ headerShown: false }}
        name='Pin'
        component={PinScreen}
      />
      <Stack.Screen
        name='Login'
        component={LoginScreen}
        options={{
          headerTitleAlign: 'left',
          title: translate('TEXT_LOG_IN')
        }}
      />
    </Stack.Navigator>
  );
}
