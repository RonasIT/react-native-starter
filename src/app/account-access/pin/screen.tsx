import { RouteProp } from '@react-navigation/native';
import React, { ReactElement, useEffect, useState } from 'react';
import { LocalAuthActions } from '@shared/local-auth';
import { localAuthService } from '@shared/local-auth/service';
import { PinForm } from '@shared/pin-form/component';
import { storeRef } from '@store';
import { AccountAccessNavigationParams } from '../navigation';

const pinLength = 4;

export function PinScreen(props: { route: RouteProp<AccountAccessNavigationParams, 'Pin'> }): ReactElement {
  const isPinSet = props?.route?.params?.isPinSet;
  const [inputValue, setInputValue] = useState('');
  const [confirmPin, setConfirmPin] = useState('');
  const [hasError, setHasError] = useState(false);
  const confirmMode = !isPinSet && inputValue.length === pinLength;

  const checkPinCode = async (): Promise<void> => {
    const isCorrect = await localAuthService.checkPin(inputValue);
    isCorrect ? storeRef.dispatch(LocalAuthActions.localAuthSuccess()) : setHasError(true);
  };

  const handleError = (): void => {
    setInputValue('');
    setConfirmPin('');
    setHasError(false);
  };

  useEffect(() => {
    if (inputValue.length === pinLength && isPinSet) {
      checkPinCode();
    }
    if (confirmMode && confirmPin.length === pinLength) {
      confirmPin === inputValue ? storeRef.dispatch(LocalAuthActions.setPin({ pin: confirmPin })) : setHasError(true);
    }
  }, [inputValue, confirmPin]);

  return (
    <PinForm
      hasError={hasError}
      handleError={handleError}
      pinLength={pinLength}
      value={confirmMode ? confirmPin : inputValue}
      onValueChange={confirmMode ? setConfirmPin : setInputValue}
      title={isPinSet ? 'Enter Pin' : confirmMode ? 'Confirm Pin' : 'Set Pin'}
      isPinSet={isPinSet}
    />
  );
}
