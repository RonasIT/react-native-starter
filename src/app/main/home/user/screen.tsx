import { RouteProp } from '@react-navigation/native';
import { useFormik } from 'formik';
import React, { useEffect } from 'react';
import { View } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { HomeNavigationParams } from '@app/main/home/navigation';
import { AppButton } from '@shared/button';
import { useTranslation } from '@shared/i18n';
import { InputFormGroup } from '@shared/input-form-group';
import { AppScreen } from '@shared/screen';
import { AppText, TextTheme } from '@shared/text';
import { userAPI } from '@shared/user/api';
import { commonStyle, createStyles } from '@styles';
import { UserForm } from './shared/forms';

export function UserScreen(props: { route: RouteProp<HomeNavigationParams, 'User'> }): JSX.Element {
  const id = props.route.params?.id;
  const translate = useTranslation('MAIN.USER');
  const { useLazyGetQuery } = userAPI;
  const [loadUser, { data: user }] = useLazyGetQuery();

  useEffect(() => {
    if (id) {
      loadUser({ id });
    }
  }, []);

  useEffect(() => {
    if (user) {
      formik.setValues({
        email: user.email,
        name: user.name
      });
    }
  }, [user]);

  function formSubmitted(values: UserForm): void {
    console.log(values);
  }

  function deleteButtonPressed(): void {
    console.log('Delete');
  }

  const formik = useFormik({
    initialValues: new UserForm(),
    validationSchema: UserForm.validationSchema,
    onSubmit: formSubmitted
  });
  const { handleSubmit } = formik;

  return (
    <KeyboardAwareScrollView>
      <AppScreen style={commonStyle.container}>
        <AppText theme={TextTheme.LARGE} style={style.title}>
          {id ? translate('TEXT_EDIT') : translate('TEXT_CREATE')}
        </AppText>
        <InputFormGroup<UserForm>
          label={translate('TEXT_EMAIL')}
          name='email'
          autoCapitalize='none'
          keyboardType='email-address'
          formik={formik}
        />
        <InputFormGroup
          label={translate('TEXT_NAME')}
          name='name'
          formik={formik} />
        <View style={style.footer}>
          {id && (
            <AppButton
              title={translate('BUTTON_DELETE')}
              onPress={deleteButtonPressed}
              style={[style.button, style.deleteButton]}
              theme='secondary'
            />
          )}
          <AppButton
            isDisabled={!formik.isValid && formik.submitCount > 0}
            title={translate('BUTTON_SAVE')}
            onPress={() => handleSubmit()}
            style={style.button}
          />
        </View>
      </AppScreen>
    </KeyboardAwareScrollView>
  );
}

const style = createStyles({
  title: {
    marginVertical: '2rem'
  },
  footer: {
    marginTop: '1.5rem',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  button: {
    flex: 1
  },
  deleteButton: {
    marginRight: '1rem'
  }
});
