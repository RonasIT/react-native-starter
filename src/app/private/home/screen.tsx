import { homeScreenFacade } from '@app/private/home/facade';
import { ItemsList } from '@shared/items-list';
import { AppScreen } from '@shared/screen';
import { User } from '@shared/user';
import { commonStyle } from '@styles';
import React, { ReactElement, useEffect } from 'react';
import { HomeListItem } from './components';

export function HomeScreen(): ReactElement {
  const { items, isLoading, isRefreshing, pagination } = homeScreenFacade;
  const refreshItems = (): void => homeScreenFacade.refreshItems();
  const loadMore = (): void => homeScreenFacade.loadItems(pagination.currentPage + 1);

  useEffect(() => {
    homeScreenFacade.loadItems();
  }, []);

  return (
    <AppScreen>
      <ItemsList<User>
        data={items}
        renderItem={HomeListItem}
        isLoading={isLoading}
        isRefreshing={isRefreshing}
        canLoadMore={pagination.currentPage < pagination.lastPage}
        onEndReached={loadMore}
        onRefresh={refreshItems}
        containerStyle={commonStyle.container}
        numColumns={1}
      />
    </AppScreen>
  );
}
