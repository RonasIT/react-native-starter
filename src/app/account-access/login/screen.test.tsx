import { authService } from '@shared/auth';
import { appNavigationService } from '@shared/navigation';
import { fireEvent, render, RenderAPI, waitFor } from '@testing-library/react-native';
import { validCredentials } from '@tests/fixtures';
import { setDefaultLanguage, TestRootComponent } from '@tests/helpers';
import React from 'react';
import { ReactTestInstance } from 'react-test-renderer';
import { LoginScreen } from './screen';

describe('Login screen', () => {
  let component: RenderAPI;
  let emailInput: ReactTestInstance;
  let passwordInput: ReactTestInstance;
  let submitButton: ReactTestInstance;
  let translation: Record<string, any>;

  function initComponent(): RenderAPI {
    return render(
      <TestRootComponent>
        <LoginScreen />
      </TestRootComponent>
    );
  }

  beforeAll(() => {
    translation = setDefaultLanguage();
  });

  beforeEach(() => {
    component = initComponent();
    emailInput = component.getByTestId('email-input');
    passwordInput = component.getByTestId('password-input');
    submitButton = component.getByTestId('submit-button');
  });

  it('should match the snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have two text inputs', () => {
    expect(emailInput).toBeDefined();
    expect(passwordInput).toBeDefined();
  });

  it('should have a submit button', () => {
    expect(submitButton).toBeDefined();
  });

  it('should render validation errors by submitting the empty form', async () => {
    fireEvent.press(submitButton);

    await waitFor(() => {
      const errors = component.getAllByTestId('validation-error');
      expect(errors[0].children[0]).toBe(translation.COMMON.VALIDATION.TEXT_VALIDATION_REQUIRED_FIELD);
      expect(errors[1].children[0]).toBe(translation.COMMON.VALIDATION.TEXT_VALIDATION_REQUIRED_FIELD);
    });
  });

  it('should render a validation error by submitting the form with invalid email', async () => {
    fireEvent.changeText(emailInput, '123');
    fireEvent.press(submitButton);

    await waitFor(() => {
      const error = component.getAllByTestId('validation-error')[0];
      expect(error.children[0]).toBe(translation.COMMON.VALIDATION.TEXT_VALIDATION_EMAIL);
    });
  });

  it('should init authorization with valid credentials and subscribe to push notifications', async () => {
    const demoAuthorizeSpy = jest.spyOn(authService, 'demoAuthorize');
    const navigateSpy = jest.spyOn(appNavigationService, 'resetToRoute');

    fireEvent.changeText(emailInput, validCredentials.email);
    fireEvent.changeText(passwordInput, validCredentials.password);
    fireEvent.press(submitButton);

    await waitFor(
      () => {
        expect(demoAuthorizeSpy).toHaveBeenCalledWith(validCredentials);
        expect(navigateSpy).toHaveBeenCalledWith('Main');
      },
      { timeout: 7000 }
    );
  });
});
