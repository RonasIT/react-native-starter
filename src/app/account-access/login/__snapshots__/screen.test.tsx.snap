// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingHorizontal": 24,
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingTop": 32,
        }
      }
    >
      <View>
        <Image
          accessibilityRole="image"
          accessible={false}
          assetGroup="icons"
          onError={[Function]}
          source={1}
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              false,
              {
                "height": 56,
                "width": 56,
              },
              false,
            ]
          }
        />
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "color": "#20303C",
                "textAlign": "left",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              [
                {
                  "fontFamily": "SFProDisplayBold",
                  "fontSize": 27.2,
                  "lineHeight": 39.36,
                },
                {
                  "color": "#FFFFFF",
                },
                {
                  "marginVertical": 32,
                },
              ],
            ]
          }
        >
          Welcome to [missing "{{value}}" value]
        </Text>
        <View
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              [
                {},
                undefined,
                {
                  "marginBottom": 19.2,
                },
              ],
            ]
          }
        >
          <Text
            color="#20303C"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                {
                  "color": "#20303C",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "minHeight": 20,
                  },
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "SFProDisplayRegular",
                    "fontSize": 22.4,
                    "lineHeight": 24.64,
                    "marginBottom": 8,
                  },
                  undefined,
                ],
              ]
            }
            testID="email-input.label"
          >
            Email
          </Text>
          <View
            centerV={true}
            row={true}
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                {},
                {},
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                [
                  {},
                  [
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#18191F",
                        "borderColor": "#18191F",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "position": "relative",
                        "width": "100%",
                      },
                      false,
                      false,
                      undefined,
                      undefined,
                      undefined,
                    ],
                    undefined,
                  ],
                ],
              ]
            }
          >
            <View
              flexG={true}
              style={
                [
                  undefined,
                  undefined,
                  {
                    "flexGrow": 1,
                  },
                  undefined,
                  {},
                  {},
                  {},
                  undefined,
                ]
              }
            >
              <TextInput
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                autoCapitalize="none"
                editable={true}
                keyboardType="email-address"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                style={
                  [
                    {
                      "flexGrow": 1,
                      "paddingBottom": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                    },
                    {
                      "color": "#20303C",
                    },
                    [
                      {},
                      undefined,
                      [
                        {
                          "color": "#FFFFFF",
                          "flex": 1,
                          "fontSize": 18.4,
                          "height": 54,
                          "paddingBottom": 15,
                          "paddingTop": 15,
                        },
                        {},
                        undefined,
                      ],
                    ],
                  ]
                }
                testID="email-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
          <View
            row={true}
            spread={true}
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                {},
                {},
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <Text
              $textDangerLight={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#20303C",
                    "textAlign": "left",
                  },
                  undefined,
                  {
                    "color": "#FC3D2F",
                  },
                  undefined,
                  undefined,
                  {},
                  undefined,
                  undefined,
                  undefined,
                  [
                    {
                      "minHeight": 20,
                    },
                    {
                      "color": "#FF003D",
                      "fontFamily": "SFProTextRegular",
                      "fontSize": 16,
                      "lineHeight": 29.44,
                    },
                  ],
                ]
              }
              testID="email-input.validationMessage"
            />
          </View>
        </View>
        <View
          style={
            [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              [
                {},
                undefined,
                {
                  "marginBottom": 19.2,
                },
              ],
            ]
          }
        >
          <Text
            color="#20303C"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                {
                  "color": "#20303C",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                [
                  {
                    "minHeight": 20,
                  },
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "SFProDisplayRegular",
                    "fontSize": 22.4,
                    "lineHeight": 24.64,
                    "marginBottom": 8,
                  },
                  undefined,
                ],
              ]
            }
            testID="password-input.label"
          >
            Password
          </Text>
          <View
            centerV={true}
            row={true}
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                {},
                {},
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                [
                  {},
                  [
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#18191F",
                        "borderColor": "#18191F",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "paddingLeft": 20,
                        "paddingRight": 20,
                        "position": "relative",
                        "width": "100%",
                      },
                      false,
                      false,
                      {
                        "paddingRight": 50,
                      },
                      undefined,
                      undefined,
                    ],
                    undefined,
                  ],
                ],
              ]
            }
          >
            <View
              flexG={true}
              style={
                [
                  undefined,
                  undefined,
                  {
                    "flexGrow": 1,
                  },
                  undefined,
                  {},
                  {},
                  {},
                  undefined,
                ]
              }
            >
              <TextInput
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                secureTextEntry={true}
                style={
                  [
                    {
                      "flexGrow": 1,
                      "paddingBottom": 0,
                      "paddingTop": 0,
                      "textAlign": "left",
                    },
                    {
                      "color": "#20303C",
                    },
                    [
                      {},
                      undefined,
                      [
                        {
                          "color": "#FFFFFF",
                          "flex": 1,
                          "fontSize": 18.4,
                          "height": 54,
                          "paddingBottom": 15,
                          "paddingTop": 15,
                        },
                        {},
                        undefined,
                      ],
                    ],
                  ]
                }
                testID="password-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0.5,
                  "position": "absolute",
                  "right": 20,
                  "top": 0,
                  "width": 30,
                }
              }
            />
          </View>
          <View
            row={true}
            spread={true}
            style={
              [
                undefined,
                undefined,
                undefined,
                undefined,
                {},
                {},
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <Text
              $textDangerLight={true}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#20303C",
                    "textAlign": "left",
                  },
                  undefined,
                  {
                    "color": "#FC3D2F",
                  },
                  undefined,
                  undefined,
                  {},
                  undefined,
                  undefined,
                  undefined,
                  [
                    {
                      "minHeight": 20,
                    },
                    {
                      "color": "#FF003D",
                      "fontFamily": "SFProTextRegular",
                      "fontSize": 16,
                      "lineHeight": 29.44,
                    },
                  ],
                ]
              }
              testID="password-input.validationMessage"
            />
          </View>
        </View>
        <View
          style={
            {
              "marginTop": 32,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#26a0f8",
                "borderRadius": 12,
                "flexDirection": "row",
                "justifyContent": "center",
                "minWidth": 90,
                "opacity": 1,
                "paddingHorizontal": 15,
                "paddingVertical": 13,
                "width": "100%",
              }
            }
            testID="submit-button"
          >
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "color": "#20303C",
                    "textAlign": "left",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  [
                    {
                      "backgroundColor": "transparent",
                      "flex": 0,
                      "flexDirection": "row",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    {
                      "color": "#FFFFFF",
                    },
                    undefined,
                    undefined,
                    [
                      {
                        "fontFamily": "SFProTextSemiBold",
                        "fontSize": 18.4,
                        "fontWeight": "600",
                        "lineHeight": 30,
                        "textAlign": "center",
                      },
                      {
                        "color": "#FFFFFF",
                      },
                      false,
                    ],
                  ],
                ]
              }
              testID="submit-button.label"
            >
              Login
            </Text>
          </View>
        </View>
        <Text
          style={
            [
              {
                "backgroundColor": "transparent",
                "color": "#20303C",
                "textAlign": "left",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              [
                {
                  "fontFamily": "SFProTextRegular",
                  "fontSize": 16,
                  "lineHeight": 29.44,
                },
                {
                  "color": "#FFFFFF",
                },
                [
                  {
                    "fontSize": 13.6,
                    "textAlign": "right",
                    "width": "100%",
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          vundefined (undefined)
        </Text>
        <KeyboardTrackingViewTemp
          pointerEvents="none"
          scrollToFocusedInput={true}
          style={
            {
              "backgroundColor": "transparent",
              "bottom": 0,
              "height": 0.5,
              "left": 0,
              "position": "absolute",
              "width": 750,
            }
          }
          useSafeArea={false}
        />
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
